interpreter/llvm/src/tools/clang/lib/Basic/Diagnostic.cpp

```diff
@@ -172,7 +172,29 @@ void DiagnosticsEngine::DiagStateMap::append(SourceManager &SrcMgr,
        Offset = F->ParentOffset, F = F->Parent) {
     F->HasLocalTransitions = true;
     auto &Last = F->StateTransitions.back();
-    assert(Last.Offset <= Offset && "state transitions added out of order");
+    if (Last.Offset > Offset) {
+      // Deal with a state change induce by recursive parsing.  The first parsing is
+      // suspended and a (recursive) parsing is started between associated (in the upper/outer
```

Comment:
"is started between associated" doesn't read well. Which I really mean as a euphemism because I cannot parse that :-)